File: 151002/Kragel/lab1/src/main/java/by/bsuir/dc/entity/Tweet.java
package by.bsuir.dc.entity;

import lombok.Getter;
import lombok.Setter;

import java.time.Instant;

@Getter
@Setter
public class Tweet extends AbstractEntity{
    private Long creatorId;
    private String title;
    private String content;
    private Instant created;
    private Instant modified;
}

File: 151004/Buhovets/lab1/REST/Entity/Db/Tweet.cs
using REST.Entity.Common;
using System.ComponentModel.DataAnnotations;

namespace REST.Entity.Db
{
    public class Tweet(string title, string content, DateTime created, DateTime modified) : AbstractEntity
    {
        [MinLength(2)]
        public string Title { get; set; } = title;
        [MinLength(4)]
        public string Content { get; set; } = content;
        public DateTime Created { get; set; } = created;
        public DateTime Modified { get; set; } = modified;
        public ICollection<Post> Posts { get; set; } = [];
        public ICollection<Sticker> Markers { get; set; } = [];
        public Creator Creator { get; set; } = null!;

        public Tweet() : this(string.Empty, string.Empty, DateTime.Now, DateTime.Now) { }
        public Tweet(int id, Creator creator, string title, string content, DateTime created, DateTime modified)
            : this(title, content, created, modified)
        {
            Id = id;
            Creator = creator;
        }
    }
}
